File: app/src/main/java/com/lsy/webviewassembly/KingoWebViewApplication.java
package com.lsy.webviewassembly;

import com.kingja.loadsir.core.LoadSir;
import com.lsy.kingobase.BaseApplication;
import com.lsy.kingobase.loadsir.CustomCallback;
import com.lsy.kingobase.loadsir.EmptyCallback;
import com.lsy.kingobase.loadsir.ErrorCallback;
import com.lsy.kingobase.loadsir.LoadingCallback;
import com.lsy.kingobase.loadsir.TimeoutCallback;

public class KingoWebViewApplication extends BaseApplication {

    @Override
    public void onCreate(){
        super.onCreate();
        LoadSir.beginBuilder()
                .addCallback(new ErrorCallback())
                .addCallback(new EmptyCallback())
                .addCallback(new LoadingCallback())
                .addCallback(new TimeoutCallback())
                .addCallback(new CustomCallback())
                .setDefaultCallback(LoadingCallback.class)
                .commit();
    }
}

File: app/src/main/java/com/lsy/webviewassembly/KingoWebViewApplication.kt
package com.lsy.webviewassembly

import com.kingja.loadsir.core.LoadSir
import com.lsy.kingobase.BaseApplication
import com.lsy.kingobase.loadsir.*

class KingoWebViewApplication : BaseApplication() {
    override fun onCreate() {
        super.onCreate()
        //初始化LoadSir
        LoadSir.beginBuilder()
            .addCallback(ErrorCallback()) //加载失败回调处理
            .addCallback(EmptyCallback()) //空回调处理
            .addCallback(LoadingCallback()) //加载回调处理
            .addCallback(TimeoutCallback()) //超时回调处理
            .addCallback(CustomCallback()) //自定义回调处理
            .addCallback(AnimateCallback())//自定义动画处理
            .setDefaultCallback(LoadingCallback::class.java) //默认回调处理
            .commit()
    }
}